pkg/servicedisc/client.go

```diff
@@ -269,40 +269,10 @@ func (c *HTTPClient) DeleteEntry(ctx context.Context) (err error) {
 	return nil
 }
 
-// UpdateLoop repetitively calls 'POST /api/services' to update entry.
-func (c *HTTPClient) UpdateLoop(ctx context.Context, updateInterval time.Duration) {
-	defer func() { _ = c.DeleteEntry(context.Background()) }() //nolint:errcheck
-
-	ticker := time.NewTicker(updateInterval)
-	defer ticker.Stop()
-
-	for {
-		if err := c.Update(ctx); errors.Is(err, ErrVisorUnreachable) {
-			return
-		}
-		c.entryMx.Lock()
-		j, err := json.Marshal(c.entry)
-		c.entryMx.Unlock()
-		logger := c.log.WithField("entry", string(j))
-		if err == nil {
-			logger.Debug("Entry updated.")
-		} else {
-			logger.Errorf("Service returned malformed entry, error: %s", err)
-			return
-		}
-
-		select {
-		case <-ctx.Done():
-			return
-		case <-ticker.C:
-		}
-	}
-}
-
-// Update calls 'POST /api/services' to update service discovery entry
-// it performs exponential backoff in case of errors during update, unless
+// RegisterEntry calls 'POST /api/services' to register service discovery entry
+// it performs exponential backoff in case of errors during register, unless
 // the error is unrecoverable from
-func (c *HTTPClient) Update(ctx context.Context) error {
+func (c *HTTPClient) RegisterEntry(ctx context.Context) error {
```

Comment:
We should name this `Register`

Reply:
@jdknives I think it would be better if we are consistent with the naming convention. We are using `entry` here on the client side but `service` on `skycoin-service-discovery` side.

Reply:
Done.